creed/tiling_utils.py

```diff
@@ -0,0 +1,1115 @@
+        )
+
+        size_y = read_tiling_param(
+            "Enter the y step size for region %s (in microns): " % fov['name'],
+            "Error: y step size must be a positive integer",
+            lambda sy: sy >= 1,
+            dtype=int
+        )
+
+        region_params['x_fov_size'].append(size_x)
+        region_params['y_fov_size'].append(size_y)
+
+        # allow the user to specify if the FOVs should be randomized
+        randomize = read_tiling_param(
+            "Randomize FOVs for region %s? Y/N: " % fov['name'],
+            "Error: randomize parameter must Y or N",
+            lambda r: r in ['Y', 'N', 'y', 'n'],
+            dtype=str
+        )
+
+        randomize = randomize.upper()
+
+        region_params['region_rand'].append(randomize)
+
+
+def set_tiled_region_params(region_corners_path):
+    """Given a file specifying FOV regions, set the MIBI tiling parameters
+
+    User inputs will be required for many values.
+
+    Args:
+        region_corners_path (str):
+            Path to the JSON file containing the FOVs used to define the upper-left corner
+            of each tiled region
+
+    Returns:
+        dict:
+            Contains the tiling parameters for each tiled region
+    """
+
+    # file path validation
+    if not os.path.exists(region_corners_path):
+        raise FileNotFoundError(
+            "Tiled region corners list file %s does not exist" % region_corners_path
+        )
+
+    # read in the region corners data
+    with open(region_corners_path, 'r') as flf:
+        tiled_region_corners = json.load(flf)
+
+    # define the parameter dict to return
+    tiling_params = {}
+
+    # copy over the metadata values from tiled_region_corners to tiling_params
+    tiling_params = assign_metadata_vals(tiled_region_corners, tiling_params, ['fovs'])
+
+    # define the region_params dict
+    region_params = {rpf: [] for rpf in settings.REGION_PARAM_FIELDS}
+
+    # prompt the user for params associated with each tiled region
+    read_tiled_region_inputs(tiled_region_corners, region_params)
+
+    # need to copy fov metadata over, needed for generate_fov_list
+    tiling_params['fovs'] = copy.deepcopy(tiled_region_corners['fovs'])
+
+    # store the read in parameters in the region_params key
+    tiling_params['region_params'] = generate_region_info(region_params)
+
+    # whether to insert moly points between regions
+    moly_region_insert = read_tiling_param(
+        "Insert moly points between regions? Y/N: ",
+        "Error: moly point region parameter must be either Y or N",
+        lambda mri: mri in ['Y', 'N', 'y', 'n'],
+        dtype=str
+    )
+
+    # convert to uppercase to standardize
+    moly_region_insert = moly_region_insert.upper()
+    tiling_params['moly_region'] = moly_region_insert
+
+    # whether to insert moly points between fovs
+    moly_interval = read_tiling_param(
+        "Enter the FOV interval size to insert Moly points. If you do not want to insert "
```

Comment:
Change to: "Insert moly points between FOVs? If yes, enter the number of FOVs between each Moly point. If no, enter 0. "